os/fs/smartfs/smartfs_utils.c

```diff
@@ -2029,9 +2029,10 @@ int smartfs_scan_entry(struct smartfs_mountpt_s *fs, char *map, struct sector_re
 			/* Read all of entries in current sector */
 			smartfs_setbuffer(&readwrite, logsector, 0, fs->fs_llformat.availbytes, (uint8_t *)fs->fs_rwbuffer);
 			ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
+			/* If current sector is not valid, move to next sector */
 			if (ret < 0) {
 				fdbg("Error %d reading sector %d data\n", ret, logsector);
-				goto errout;
+				break;
```

Comment:
Line 2161: - `ret = OK;`
won't this line reset ret = OK before the function exits?

Reply:
yes, it will pop next node and will test this again in main loop, while (node != NILL)

Reply:
I mean earlier it could return with ret<0 by going to label "errout"
But now ret will always be assigned = OK before it reaches label errout?

Reply:
For stable recovery, If read is failed, it will continue to recovery because read doesn't change contents on flash.
But another ioctl include BIOC_WRITESECT, it can affect to contents on flash, so it will go to errout.

Reply:
Ohkkay.
Then it seems alright to continue with recovery instead of exiting in case of read failure.